app/presenters/vendor_api/single_application_presenter.rb

```diff
@@ -255,6 +261,59 @@ def qualification_to_hash(qualification)
       }.merge HesaQualificationFieldsPresenter.new(qualification).to_hash
     end
 
+    def subject_code(qualification)
+      # There are a few cases here:
+      #
+      # Qualification level: gcse
+      #   These are either maths, science or english GCSEs that are required by some courses
+      #   - Maths GCSEs are not structured, so we map to a single maths subject code
+      #   - Science GCSEs are structured, but we combine them into one in the API response, so we can only be as granular as 'science'
+      #   - English GCSEs are now structured, but they did not used to be. We map to a generic english code here for the unstructured ones,
+      #     and have a better mapping within the structured_gcse_to_hashes method for newer english GCSEs
+      #
+      # Qualification level: other
+      #   These are other qualifications that a candidate has that are not direct requirements for the course.
+      #   This includes GCSEs and A levels, and none of them are structured. If their subject has been chosen
+      #   directly from the autocomplete, then we can map to a code. Otherwise, we leave the code blank.
+
+      if qualification.gcse?
+        subject_code_for(qualification.subject)
+      elsif qualification.other?
+        subject_code_for_other_qualification(qualification)
+      end
+    end
+
+    def subject_code_for(subject)
+      case subject
+      when 'maths'
+        '100403'
+      when ApplicationQualification::SCIENCE, ApplicationQualification::SCIENCE_SINGLE_AWARD, ApplicationQualification::SCIENCE_DOUBLE_AWARD, ApplicationQualification::SCIENCE_TRIPLE_AWARD
+        '100390'
+      when 'english', 'english_single_award', 'english_double_award', 'english_studies_single_award', 'english_studies_double_award'
+        '100320'
+      when 'english_language'
+        '100318'
+      when 'english_literature'
+        '100319'
+      end
+    end
+
+    def subject_code_for_other_qualification(qualification)
```

Comment:
I think we can simplify this method further. Maybe if we defined the constants as the key being the name of the qualification and the code being the hecos code, maybe we can simplify the method as so:
```
if qualification.qualification_type == 'GCSE'
  GCSE_SUBJECTS[qualification.subject]
elsif ['A level', 'AS level'].include? qualification.qualification_type
  A_AND_AS_LEVEL_SUBJECTS[qualification.subject]
end
```
If not we can still avoid returning many times in the method